src/app/services/ApiService.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect } from 'react';
+
+import { ipcBridge } from '@bridge';
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+
+import { useQueue } from '@app/utils';
+import { IPC_CHANNELS, JsonRPCRequest, JsonRPCResponse } from '@types';
+
+export function useApiService() {
```

Comment:
SOC: ApiService responsibility is only to talk with the ipcBridge. 
Queue (handling the app state) is a different concern.

Reply:
Will solve in a later PR.